### PR TITLE
CSI trim methods rewriting

### PR DESCRIPTION
**Pull request by iunanua**
### What does this PR do?
Enables `trim`, `trimStart` and `trimEnd` methods rewriting

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

**Comment by uurien** [comment]
I think that we need more integration test, we can copy and rewrite some files and exec it in the requests to create some integration test.

File: packages/dd-trace/src/appsec/iast/taint-tracking/taint-tracking-impl.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+function getCsiFn (cb, ...protos) {
+  let filter
+  if (!protos || protos.length === 0) {
+    filter = (res, fn, target) => notString(res, target)
+  } else if (protos.length === 1) {
+    const protoFn = protos[0]
+    filter = (res, fn, target) => notString(res, target) || fn !== protoFn
+  } else {
+    filter = (res, fn, target) => notString(res, target) || !isValidCsiMethod(fn, ...protos)
```
Comment (uurien) [inline packages/dd-trace/src/appsec/iast/taint-tracking/taint-tracking-impl.js:52]: you are converting array to params, to convert params to array in line 40.
Why not something like: `!isValidCsiMethod(fn, protos)`
And change line 40 to:
```
function isValidCsiMethod (fn, ...protos) {
```

File: packages/dd-trace/test/appsec/iast/utils.js [hunk 2]
```diff
@@ -106,4 +112,12 @@ function testThatRequestHasNotVulnerability (app, vulnerability) {
   testInRequest(app, tests)
 }
 
-module.exports = { testThatRequestHasNotVulnerability, testThatRequestHasVulnerability, testInRequest }
+function copyFileToTmp (src) {
+  const prefix = Math.floor(Math.random() * 1000000)
```
Comment (uurien) [inline packages/dd-trace/test/appsec/iast/utils.js:116]: If you create a random prefix out of the method, a here, only increase a counter, you'll have a prefix imposible to duplicate in the same execution.
I know that is nearly imposible, but you can have same Math.random() result, but you can't have same id if you increment a counter.

File: packages/dd-trace/test/appsec/iast/taint-tracking/taint-tracking-impl.spec.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      describe(`using ${propFn}()`, () => {
+        commands.forEach((command) => {
+          describe(`with command: '${command}'`, () => {
+            testThatRequestHasVulnerability(function () {
+              const store = storage.getStore()
+              const iastContext = iastContextFunctions.getIastContext(store)
+              const commandTainted = newTaintedString(iastContext, command, 'param', 'Request')
+
+              const propFnInstrumented = require(instrumentedFunctionsFile)[propFn]
+              const propFnOriginal = require(propagationFunctionsFile)[propFn]
```
Comment (uurien) [inline packages/dd-trace/test/appsec/iast/taint-tracking/taint-tracking-impl.spec.js:56]: Can we do this require in line 33? At this way we will require it only once and out of the request/iast context.

**Review by hoolioh — APPROVED** [review]

**Review by Qard — APPROVED** [review]

**Review by uurien — APPROVED** [review]

**Review by uurien — APPROVED** [review]

**Pull request merged**